tests/test-resource-groups/main.tf

```
@@ -0,0 +1,12 @@
+provider "aws" {
+  region = "us-east-1"
+}
+
+resource "aws_resourcegroups_group" "test" {
+  name = var.name
+
+  resource_query {
+    query = jsonencode(var.resource_query)
+    type  = var.type
+  }
+}
```

The test config should be a `module` that exercise the terraform code in this project.

```suggestion
module "resource_group" {
  name = ...
  type = ...
  resource_query = ...
```

---

resolved. I had tried the module first, but it kept giving me an error about the resource query, i realized it wanted an `argument` not a `resource_query`